data-contract/src/contract/invalidate.rs

```diff
@@ -105,11 +107,15 @@ impl CertificationContract {
         // Remove from owners map
         self.tokens.owner_by_id.remove(&token_id);
 
+        let empty_account_id: AccountId = unsafe {
+            std::mem::transmute(UnsafeAccountId(String::new()))
+        };
+
```

Comment:
@ryancwalsh would appreciate input on this strategy.

Since a standard "NFT Deletion" event doesn't exist, what I'm doing is emitting an NFT transfer event, but the new owner ID is an empty string, so hopefully the wallet will not pick it up (or at least not display it on any accounts).

`unsafe` Rust code is usually frowned upon, because it allows the code to violate many of the safety checks that Rust is so well-known for. In this case, all I'm doing is reinterpreting a custom struct (with identical members as `AccountId`) as an `AccountId`, because none of the public methods (including `new_unchecked` at least in debug mode) allow me to create an instance that violates https://docs.near.org/docs/concepts/account#account-id-rules

Since I'm intentionally violating the NFT event standard here, I think it warrants violating the safety checks that would enforce the standard as well. Thoughts?

Reply:
@petarvujovic98 you too, since you're more familiar with Rust

Reply:
@encody Maybe it would be easier to discuss this live. My assumption was that you'd be coding a `delete_cert` function (or whatever you want to call it) that would probably be a simple wrapper of:

1. calling the function that invalidates the cert 
2. calling the transfer function, with the trash bin account as the recipient

Something like that. Given my perhaps naive guess about what I was asking for, it didn't feel like it was breaking standards. It felt like it was a workaround where instead of letting real people's wallets stay cluttered with invalidated certs, the clutter would be moved to a wallet that nobody cares about.

I'm not sure what `std::mem::transmute(UnsafeAccountId(String::new()))` does, but I was surprised to see that this PR required so much new code.

Reply:
@encody Oh ok, you're talking about `#2` from https://github.com/NEAR-Edu/near-certification-tools/issues/52.

To be honest I still am not clear on why "unsafe" code is needed.

Why do you need to violate https://docs.near.org/docs/concepts/account#account-id-rules?

Reply:
@ryancwalsh An empty string is less than two characters in length.

Reply:
@encody  Yep, but I mean why can't we "send" the NFT (instead of to an empty string) to a valid account ID but one that is inert (either all zeroes or some specific implicit account we create that nobody cares about)?

Reply:
@ryancwalsh 

That's what the trash account is for, and will be used when invalidating the certificates. The issue that this PR addresses also [requests](https://github.com/NEAR-Edu/near-certification-tools/issues/52#issue-1208353313):

> Can we also have a way of deleting (not just invalidating) certs? This would be used very sparingly (probably just pre-launch to clean up all the clutter that Petar and I created on mainnet).

Simply deleting the data from the contract doesn't remove the event that was emitted when the NFT was first minted / transferred, so this ensures that there is an assuredly unusable NFT transfer event emitted as a last-ditch effort to make sure the bad data won't affect anything.

Reply:
@encody  I'm still not understanding, so we'll talk about it live. I'll reschedule our 1-on-1.